### PR TITLE
bt-638: Company talent hiring info

### LINKED ISSUE #638: Company talent hiring info
We need implement logic to hiring info.

Backend:
- [x] create relation table with companyId => userDetailsTable, talentId => userDetailsTable and other columns which we provide while hiring
- [x] create repository with create method and method to get all records about hiring
- [x] create service with such method
- [x] create endpoint GET and POST

Frontend:
- [x] implement hiring process (talent, company)
- [x] add list of hiring record in admin layout (Maybe add another page for this) - https://mui.com/material-ui/react-table/

Screenshots: 
Image

Image

**Please clarify hiring process with QA team !!!**

### LINKED ISSUE #670: Add logic to Sign in flow when the profile is saved in drafts/published/approved by admin
**Acceptance criteria:**

- When talent signs in and the profile is saved without publishing ("Save without publishing" button) he is redirected to step 5 - Preview page
- When the talent signs in and the profile is published ("Publish now" button) he is redirected to step 5 - Preview page
- When the talent's profile is approved by the admin he is redirected to the Chat screen

### PR DESCRIPTION
**Pull request by savinaDaria**
image
image
image
image
image

### CODE REVIEW

**Comment by nikita-remeslov** [comment]
image
looks like our verification page should be Home page.
We can add another icon in the sidebar and call it Connection page

**Comment by nikita-remeslov** [comment]
Also sidebar design. I think @JosuerBague did something with admin sidebar

File: backend/src/migrations/20230919162409_update_bsa_badges_table.ts
Comment (nikita-remeslov) [inline backend/src/migrations/20230919162409_update_bsa_badges_table.ts:1]: we should create new migrations to add those changes!

File: backend/src/migrations/20230907092959_update_files_table.ts
Comment (nikita-remeslov) [inline backend/src/migrations/20230907092959_update_files_table.ts:1]: the same

File: backend/src/migrations/20230923202930_add_hiring_info_table.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import { type Knex } from 'knex';
+
+const uuid = 'uuid_generate_v4()';
```
Comment (nikita-remeslov) [inline backend/src/migrations/20230923202930_add_hiring_info_table.ts]:
```suggestion
const UUID = 'uuid_generate_v4()';
```

File: backend/src/migrations/20230923202930_add_hiring_info_table.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+const constraintName = 'hiring_info_pkey';
```
Comment (nikita-remeslov) [inline backend/src/migrations/20230923202930_add_hiring_info_table.ts]: UPPERCASE

File: backend/src/migrations/20230923202930_add_hiring_info_table.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+const TableName = {
```
Comment (nikita-remeslov) [inline backend/src/migrations/20230923202930_add_hiring_info_table.ts:6]: enum

File: backend/src/migrations/20230923202930_add_hiring_info_table.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    HIRING_INFO: 'hiring_info',
+    USER_DETAILS: 'user_details',
+};
+
+const ColumnName = {
```
Comment (nikita-remeslov) [inline backend/src/migrations/20230923202930_add_hiring_info_table.ts:11]: enum

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

**Comment by savinaDaria** [comment]
I also added Home component to handle navigation for different roles in Root

**Review by nikita-remeslov — APPROVED** [review]

File: backend/src/migrations/20230923202930_add_hiring_info_table.ts [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+async function up(knex: Knex): Promise<void> {
+    await knex.raw('CREATE EXTENSION IF NOT EXISTS "uuid-ossp";');
+
+    return knex.schema.createTable(TableName.HIRING_INFO, (table) => {
+        table
+            .uuid(ColumnName.ID)
+            .unique()
+            .notNullable()
+            .defaultTo(knex.raw(UUID))
+            .primary(CONTRAINT_NAME);
```
Comment (nika-sirak) [inline backend/src/migrations/20230923202930_add_hiring_info_table.ts]:
```suggestion
            .primary({ constraintName: CONTRAINT_NAME });
```

**Review by nika-sirak — APPROVED** [review]

**Pull request merged**